nidaqmx/_grpc_stub_interpreter.py

```diff
@@ -0,0 +1,394 @@
+import threading
+import warnings
+
+import grpc
+import nidaqmx.nidaqmx_pb2 as grpc_types
+import nidaqmx.nidaqmx_pb2_grpc as nidaqmx_grpc
+import nidaqmx.session_pb2 as session_grpc_types
+
+class GrpcStubInterpreter(object):
+    '''Interpreter for interacting with a gRPC Stub class'''
+
+    def __init__(self, grpc_options):
+        self._grpc_options = grpc_options
+        self._lock = threading.RLock()
+        self._client = nidaqmx_grpc.NiDAQmxStub(grpc_options.grpc_channel)
+        self.session_initialization_behavior = grpc_options.initialization_behavior
+        self.set_task_handle()
+        self.set_system_handle()
+        self.set_scale_handle()
+
+    def set_task_handle(self, value=session_grpc_types.Session()):
+        self._task_handle = value
+
+    def get_task_handle(self):
+        return self._task_handle
+    
+    def set_system_handle(self, value=session_grpc_types.Session()):
+        self._system_handle = value
+
+    def get_system_handle(self):
+        return self._system_handle
+
+    def set_scale_handle(self, value=session_grpc_types.Session()):
+        self._scale_handle = value
+
+    def get_scale_handle(self):
+        return self._scale_handle
```

Comment:
Scales do not have handles. They only have names. I think the scale class (not the interpreter) should store the scale name.

```
/******************************************************/
/***              Scale Configurations              ***/
/******************************************************/


int32 __CFUNC     DAQmxCreateLinScale            (const char name[], float64 slope, float64 yIntercept, int32 preScaledUnits, const char scaledUnits[]);
int32 __CFUNC     DAQmxCreateMapScale            (const char name[], float64 prescaledMin, float64 prescaledMax, float64 scaledMin, float64 scaledMax, int32 preScaledUnits, const char scaledUnits[]);
int32 __CFUNC     DAQmxCreatePolynomialScale     (const char name[], const float64 forwardCoeffs[], uInt32 numForwardCoeffsIn, const float64 reverseCoeffs[], uInt32 numReverseCoeffsIn, int32 preScaledUnits, const char scaledUnits[]);
int32 __CFUNC     DAQmxCreateTableScale          (const char name[], const float64 prescaledVals[], uInt32 numPrescaledValsIn, const float64 scaledVals[], uInt32 numScaledValsIn, int32 preScaledUnits, const char scaledUnits[]);
int32 __CFUNC     DAQmxCalculateReversePolyCoeff (const float64 forwardCoeffs[], uInt32 numForwardCoeffsIn, float64 minValX, float64 maxValX, int32 numPointsToCompute, int32 reversePolyOrder, float64 reverseCoeffs[]);

int32 __CFUNC_C   DAQmxGetScaleAttribute         (const char scaleName[], int32 attribute, void *value, ...);
int32 __CFUNC_C   DAQmxSetScaleAttribute         (const char scaleName[], int32 attribute, ...);
```

https://github.com/ni/grpc-device/blob/main/generated/nidaqmx/nidaqmx.proto#L6195
```
message CreateLinScaleRequest {
  string name = 1;
  double slope = 2;
  double y_intercept = 3;
  oneof pre_scaled_units_enum {
    UnitsPreScaled pre_scaled_units = 4;
    int32 pre_scaled_units_raw = 5;
  }
  string scaled_units = 6;
}

message CreateLinScaleResponse {
  int32 status = 1;
}
```

Reply:
I will remove the scale property from the interpreters as well.